### ISSUE TITLE
Match and overlay existing "Ahead of Print" works with newly imported "full" works

### ISSUE THREAD

**Issue by jstirnaman** [BibApp/BibApp#90]
In PubMed, citations are frequently assigned a status of "Epub Ahead of Print". This appears as PST - aheadofprint in the MEDLINE citation format.
In BibApp, if we have an existing work that is aheadofprint and we import a new full-level citation for the same work then the new full-level citation should match and overlay/replace the aheadofprint data. Ideally, we don't want to create a new work record for the new citation, but instead update and reindex the existing aheadofprint record.
Is it possible without modifying the works schema?

Possible way forward:

If "PST - aheadofprint" exists in incoming MEDLINE citation then set one of [admin_definable, user_definable, medium] to aheadofprint. Maybe set a state instead? Should we just handle anything that has a PMID? Where does Zotero, for example, put that tag by default?
When importing a batch, if a duplicate is found that has aheadofprint in of the above fields or state then suggest updating the existing record with the new.

**Comment by hading** [comment]
I'll need to look at this in more depth. It sounds like it might be somewhat involved.

**Comment by jstirnaman** [comment]
Yes, I think so which why is I haven't devoted much time to it yet. The
overlay suggestion is mostly my aim to keep persistent links, but it's
certainly lower priority than just matching a new full citation to it's
"duplicate" AOP counterpart. The only other solution I can think of is to
implement a ID or URL alias table.

On Thu, Mar 8, 2012 at 4:14 PM, Howard Ding <
reply@reply.github.com

**Comment by jstirnaman** [comment]
I'm looking at http://ewout.name/2010/04/generic-deep-merge-for-activerecord/ as a possible solution for at least merging duplicate work objects and "Ahead of Print" stubs with the richer object. I included it as a vendor plugin and have been trying it out. Setting dupe keys, attributes and values that qualify a record as a duplicate, is easy and it addresses the problem of deep merging associations, e.g. contributorships, from a duplicate record into some defined canonical record.
It doesn't address the problem of URL persistence - redirecting deleted record URLs to the canonical record.

**Comment by jstirnaman** [comment]
Just a thought: why not assign an Authority designation on Works? Why shouldn't a work be treated just like Publications and Publishers, having a work_authority_id column in the works table? Seems to answer the #TODO in work.rb:387

**Comment by jstirnaman** [comment]
I have a feature branch that merges duplicate works: https://github.com/jstirnaman/BibApp/tree/merges-dupes
Please take a look. It needs work in places, but it's functional. The implementation in the view definitely needs some work - I just replaced the Delete method with the Merge so it's not exactly clear to the user what's going on.
